### PR TITLE
domain: Fix data race

### PR DESCRIPTION
**Pull request by zimulala**
Using MockReloadFailed  leads to data race. Now fix it.

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -401,13 +401,34 @@ func (c *ddlCallback) OnChanged(err error) error {
 	return nil
 }
 
+// MockFailed mocks reload failed.
+// It's used for fixing data race in tests.
+// val is 1 means we need to mock reload failed.
+type MockFailed struct {
+	sync.RWMutex
+	val bool
```
Comment (ngaut) [inline domain/domain.go]: s/val/failed/

File: domain/domain.go [hunk 2]
```diff
@@ -401,13 +401,34 @@ func (c *ddlCallback) OnChanged(err error) error {
...
+}
+
+// SetValue sets whether we need to mock reload failed.
+func (m *MockFailed) SetValue(val bool) {
```
Comment (ngaut) [inline domain/domain.go]: s/SetValue/MarkFailed/

File: domain/domain.go [hunk 3]
```diff
@@ -401,13 +401,34 @@ func (c *ddlCallback) OnChanged(err error) error {
...
+	m.Lock()
+	defer m.Unlock()
+	m.val = val
+}
+
+func (m *MockFailed) getValue() bool {
```
Comment (ngaut) [inline domain/domain.go]: s/getValue/isFailed/

**Comment by coocood** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**